pyvista/core/datasetattributes.py

```diff
@@ -547,8 +544,18 @@ def get_array(
             if type(vtk_arr) == _vtk.vtkAbstractArray:
```

Comment:
Tests fail when changing to `isinstance`:
```

    @active_scalars_name.setter
    def active_scalars_name(self, name: str) -> None:
        self._raise_field_data_no_scalars_vectors()
>       dtype = self[name].dtype
E       AttributeError: 'vtkmodules.vtkCommonCore.vtkStringArray' object has no attribute 'dtype'

pyvista/core/datasetattributes.py:1112: AttributeError
```

Reply:
I see, thanks, because we're returning a VTK array instead of a numpy one, which breaks downstream.

So the question is then: do we really want to have this if? It seems likely that we're simply not hitting this branch in the test suite, and if one day we somehow end up with an vtkAbstractArray we'll get an error... At least in a local coverage report I have lying around, this branch was not taken.

The line was added in the huge `DataSetAttributes` PR so I couldn't figure out why it's there exactly...

Reply:
Of course this is also unrelated to the PR. If we want to do anything about this it will have to be another PR.

Reply:
Opened https://github.com/pyvista/pyvista/issues/2781 to track this.